phoenicis-repository/src/main/java/org/phoenicis/repository/types/GitRepository.java

```diff
@@ -65,72 +69,103 @@ private File createRepositoryLocation(String cacheDirectoryPath) {
         return new File(cacheDirectoryPath + "/git" + hashcode);
     }
 
-    @Override
-    public synchronized RepositoryDTO fetchInstallableApplications() {
-        RepositoryDTO result = null;
-        try {
-            mutex.acquire();
+    private void cloneOrUpdate() throws RepositoryException {
+        synchronized (mutex) {
             try {
                 LOGGER.info("Begin fetching process of " + this);
 
-                boolean folderExists = this.localFolder.exists();
+                boolean lockFileExists = this.lockFile.exists();
 
-                // check that the repository folder exists
-                if (!folderExists) {
-                    LOGGER.info("Creating local folder for " + this);
+                // check that the repository lock file exists
+                if (!lockFileExists) {
+                    LOGGER.info("Creating lock file for " + this);
 
-                    if (!this.localFolder.mkdirs()) {
-                        throw new RepositoryException("Couldn't create local folder for " + this);
+                    try {
+                        this.lockFile.getParentFile().mkdirs();
+                        this.lockFile.createNewFile();
+                    } catch (IOException e) {
+                        throw new RepositoryException("Couldn't create lock file " + this.lockFile.getAbsolutePath());
                     }
                 }
-                Git gitRepository = null;
+                FileOutputStream lockFileStream = new FileOutputStream(lockFile, true);
```

Comment:
I would prefer:
```java
try (FileOutputStream lockFileStream = new FileOutputStream(lockFile, true)) {
   try (FileLock lock = lockFileStream.getChannel().lock()) {
      cloneOrUpdateWithLock(lock);
   }
}
```

In addition it could be useful to replace the `FileOutputStream` and `lockFile` with a `RandomAccessFile`, which directly provides a `getChannel` method.